### PR TITLE
Add water to energy dashboard

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

Needs https://github.com/home-assistant/core/pull/80888

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/cards/energy/hui-energy-distribution-card.ts [hunk 1]
```diff
@@ -81,8 +83,9 @@ class HuiEnergyDistrubutionCard
     const hasConsumption = true;
 
     const hasSolarProduction = types.solar !== undefined;
-    const hasBattery = types.battery !== undefined;
+    const hasBattery = false; // types.battery !== undefined;
```
Comment (piitaya) [inline src/panels/lovelace/cards/energy/hui-energy-distribution-card.ts:86]:
```suggestion
    const hasBattery = types.battery !== undefined;
```
I think we must revert this one.

**Review by piitaya — CHANGES_REQUESTED** [review]

File: src/panels/lovelace/cards/energy/hui-energy-distribution-card.ts (hunk 1 above)
Comment (bramkragten) [reply src/panels/lovelace/cards/energy/hui-energy-distribution-card.ts:86]: you think 😄

File: src/components/entity/ha-statistic-picker.ts [hunk 2]
```diff
@@ -122,6 +130,19 @@ export class HaStatisticPicker extends LitElement {
...
         );
       }
+      if (includeDeviceClass) {
+        const includeDeviceClasses: (string | null)[] =
+          ensureArray(includeDeviceClass);
+        statisticIds = statisticIds.filter((meta) => {
+          const stateObj = this.hass.states[meta.statistic_id];
+          if (!stateObj) {
+            return true;
+          }
```
Comment (balloob) [inline src/components/entity/ha-statistic-picker.ts:140]: Why would we do this ? Seems like if a statistic is no longer in the state machine and so can't be checked, we don't have to show it? 

Not sure how common, we are restoring anything with a unique ID, and so state always available.

**Review by piitaya — APPROVED** [review]

**Review by balloob — APPROVED** [review]

File: src/components/entity/ha-statistic-picker.ts (hunk 2 above)
Comment (bramkragten) [reply src/components/entity/ha-statistic-picker.ts:140]: We also have external statistics

File: src/data/energy.ts [hunk 3]
```diff
@@ -642,3 +669,6 @@ export const getEnergyGasUnit = (
     ? "m³"
     : "ft³";
 };
+
+export const getEnergyWaterUnit = (hass: HomeAssistant): string | undefined =>
+  hass.config.unit_system.length === "km" ? "m³" : "ft³";
```
Comment (osos) [inline src/data/energy.ts:674]: Please consider to include "m3" and "ft3" as well as the superscript is uneasy to write for ordinary people :-)

**Pull request merged**